docs: add a section on error handling (#88)
octopize/avatar-python: doc/source/user_guide.md

## How to download an avatar dataset

This section is responsible for showing you how to download a dataset from our server.

> Note that you can't download an original dataset once it has been uploaded.

Reviewer: Explain it's for recurity reasons + no passive form
```suggestion
Note that for security reason, you can't download an original dataset once  it's uploaded.
```

   The fix is then to go look at the error message that the privacy job threw up, launch another privacy job that is successful, and launch the generation of the report with the new privacy job once it is successful.

3. `JobStatus.FAILURE`

   Jobs that fail do not throw an exception. Rather, you have to inspect the `JobStatus` that is in the `status` property.

   ```python
   job = client.jobs.get_avatarization_job(job_id)
   print(job.status) # JobStatus.FAILURE
   ```

   If the status is `JobStatus.FAILURE`, the `error_message` property will contain an explanation of the error.
   You'll have to relaunch the job again with the appropriate modifications to your input.

4. Internal error

   Internal errors happen when there is an error on the server, meaning that we did not handle the error on our side, and something unexpected happened, for which we cannot give you an exact error message.
   These come with a 500 HTTP status code, and the message is `internal error`.
   In these cases, there is not much you can do except trying again with different parameters, hoping to not trigger the error again.

>    When these happen, our error monitoring software catches these and notifies us instantly. We will reach out to you to you to help you circumvent the issue if it's possible, while we investigate on our side. We'll be hard at work trying to resolve the bug, and push out a new version with the fix.

Reviewer: Will we always reach out to them ? We don't exactly know who they are with sentry for instance...
```suggestion
   When these happen, our error monitoring software catches these and notifies us instantly. We will reach out to you to help you circumvent the issue if it's possible, while we investigate on our side. We'll be hard at work trying to resolve the bug, and push out a new version with the fix.
```

Author: The goal is to reach out to them, yes.
